scripts/ccache-clang++.sh

```diff
@@ -2,8 +2,12 @@
 if command -v ccache &> /dev/null; then
   export CCACHE_MAXSIZE=10G
   export CCACHE_CPP2=true
+  export CCACHE_DIRECT=true
+  export CCACHE_DEPEND=true
   export CCACHE_HARDLINK=true
-  export CCACHE_SLOPPINESS=file_macro,time_macros,include_file_mtime,include_file_ctime,file_stat_matches
+  export CCACHE_FILECLONE=true
+  export CCACHE_INODECACHE=true
+  export CCACHE_SLOPPINESS=file_macro,time_macros,include_file_mtime,include_file_ctime,file_stat_matches,modules,ivfsoverlay,pch_defines,system_headers
```

Comment:
It would be great with a comment linking to whatever resource you've used to determine which settings to use here.